libraries/Microsoft.Bot.Builder.Core.Extensions/ActivityFilterMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    public class ActivityFilterMiddleware : IMiddleware
+    {
+        /// <summary>
+        /// The type of Activity that the middleware should check for. e.g. ConversationUpdated or Message
+        /// </summary>
+        private readonly string _activityType;
+
+        /// <summary>
+        /// Handler to call when a matching Activity type is received
+        /// </summary>
+        private readonly ActivityFilterHandler _activityFilterHandler;
+
+        public ActivityFilterMiddleware(string activityType, ActivityFilterHandler handler)
```

Comment:
Perhaps a code comment, saying activityTypes are found in the ACtivityTypes enum.

Reply:
I was thinking about this more. 

I *THINK* the "handler" argument *SHOULD* be an IMiddleware component. That would allow composistion of these things, which seems very powerful.

Reply:
How about we add an overload, so either you can use a handler for simple scenarios (e.g. just sending an activity) or IMiddleware for more complex composition scenarios?

Reply:
Overload added with accompanying test - really nice idea @cleemullins!
Code comment re. ActivityTypes enum also added.